### PR TITLE
Emit UI event on zoom control clicks.

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Proposed Changes

Emits a ui event when zoom control is clicked including old and new workspace scale.

### Reason for Changes

Since clicking on zoom controls does not emit a workspace click event, it makes sense to emit some kind of event instead to describe what happened.

### Test Coverage

Playground

Tested on:
* Desktop Chrome

### Documentation

We may also want to update https://developers.google.com/blockly/guides/configure/web/events to include the new possible value for element (i.e. 'zoom').

### Additional Information

This PR also does some refactoring cleanup of the mouseDown event handlers for clicking on zoom controls.

### CODE REVIEW

File: core/zoom_controls.js [hunk 1]
```diff
@@ -207,16 +206,11 @@ Blockly.ZoomControls.prototype.createZoomOutSvg_ = function(rnd) {
...
 
   // Attach listener.
-  Blockly.bindEventWithChecks_(zoomoutSvg, 'mousedown', null, function(e) {
-    ws.markFocused();
-    ws.zoomCenter(-1);
-    Blockly.Touch.clearTouchIdentifier();  // Don't block future drags.
-    e.stopPropagation();  // Don't start a workspace scroll.
-    e.preventDefault();  // Stop double-clicking from selecting text.
-  });
+  Blockly.bindEventWithChecks_(
```
Comment (samelhusseini) [inline core/zoom_controls.js:212]: Not necessarily in this PR, but ideally we should be maintaining a reference to this and disposing of the event handler in the zoom control `dispose` method.

File: core/zoom_controls.js [hunk 2]
```diff
@@ -260,16 +253,28 @@ Blockly.ZoomControls.prototype.createZoomInSvg_ = function(rnd) {
...
+      zoominSvg, 'mousedown', null, this.zoom_.bind(this, 1));
+};
+
+/**
+ * Zooms in or out of the workspace.
```
Comment (samelhusseini) [inline core/zoom_controls.js:264]: I think this is too generic of a description for this method. This isn't a method for zooming in / out, it's a method that handles a mouse event for both zoom in and out buttons, perhaps something like: 

"Handles a mouse-down on the Zoom in or Zoom out buttons on the workspace."

File: core/zoom_controls.js [hunk 3]
```diff
@@ -312,19 +316,40 @@ Blockly.ZoomControls.prototype.createZoomResetSvg_ = function(rnd) {
...
-    e.stopPropagation();  // Don't start a workspace scroll.
-    e.preventDefault();  // Stop double-clicking from selecting text.
-  });
+  Blockly.bindEventWithChecks_(
+      zoomresetSvg, 'mousedown', null, this.resetZoom_.bind(this));
+};
+
+/**
+ * Resets zoom on the workspace.
+ * @param {!Event} e A mouse down event.
```
Comment (samelhusseini) [inline core/zoom_controls.js:328]: Same here,

"Handles a mouse-down on the reset zoom button on the workspace."

File: core/zoom_controls.js (hunk 1 above)
Comment (moniika) [reply core/zoom_controls.js:212]: sgtm I'll make a follow up PR to address this and add you to it

**Pull request merged**